songpal/device.py

```diff
@@ -156,7 +157,17 @@ async def get_supported_methods(self):
                 for api in service.methods:
                     # self.logger.debug("%s > %s" % (service, api))
                     if self.debug > 1:
-                        _LOGGER.debug("> %s" % api)
+                        _LOGGER.debug("> %s", api)
+                    if api.latest_supported_version is None:
+                        _LOGGER.debug(
+                            "No supported version for %s.%s, using %s",
+                            service.name,
+                            api.name,
+                            api.version,
+                        )
+                    else:
+                        if default_latest:
+                            api.use_version(api.latest_supported_version)
```

Comment:
This reads somewhat odd. Should we not default to the latest always and get rid of `default_latest` altogether?

Reply:
We could, but I wanted it to default to the existing behaviour except where a version specified. The only place I use `default_latest` is to dump the device json.

It really isn't an issue other than maybe if the parameter type checking got enabled. Which I was looking at at the time.

Reply:
Fair enough, maybe just use an `elif` to avoid  nesting, and add an else clause to raise an exception about unexpected input?

Reply:
Updated but left off the `else` as by default both are expected to be `false`.

Which makes this odd in what is essentially debug logging logic. But because we are setting a default version to match previous behaviour of using the first signature, it makes sense to do it here.

But this aligns with your other question about always send a method version, then a default method version is no longer necessary.